### ISSUE TITLE
unable to access new PWD environment in AWS: returns 404 on public IP

### ISSUE THREAD

**Issue by oskapt** [play-with-docker/play-with-docker#434]
After opening the last issue I fired up an AWS instance to just run PWD myself, but when connecting to the public IP I get a 404 - seems the playground ID is different from the outside:

```
l2         | 2020/10/15 15:29:52 Connected to network [play-with-docker_default] with ip [172.19.0.3]
pwd        | 2020/10/15 15:29:54 Updating playgrounds configuration
pwd        | 2020/10/15 15:29:54 EVENT: Playground New 4a24b940-e23f-54f5-8867-137433240dd7
pwd        | 2020/10/15 15:29:54 Updating playgrounds configuration
pwd        | 2020/10/15 15:29:54 Listening on port 3000
pwd        | [negroni] Started GET /
pwd        | 2020/10/15 15:30:12 Error retrieving playground cd803900-2037-5efd-95b4-0417224ea762. Got: NotFound
pwd        | 2020/10/15 15:30:12 Playground for domain 18.217.73.181 was not found!
pwd        | [negroni] Completed 404 Not Found in 128.002µs
```

Is this an easy fix? Some config var somewhere?

**Comment by oskapt** [comment]
There's definitely a disconnect between "go run this on your own environment" and "it only works on localhost." I've found the `xid` generator, but I'm not able to detangle how you're handling connections to localhost differently than connections to the public IP. Is the playground env that's publicly accessible running from the same codebase, or is there some other wrapper/configuration/branch that's modifying it from the default behavior outlined in the README?

**Comment by marcosnils** [comment]
> There's definitely a disconnect between "go run this on your own environment" and "it only works on localhost."

Agree. The README instructions are mostly to run it locally but not to deploy it in a cloud environment. There's been some discussions in the issues on how to accomplish this, but the main reason why we (I mostly) haven't invested a lot of time on this documentation is because not too many people request for it and I don't have the bandwidth ATM to do it. I can help you going through the steps and help with any issues along the way but as I said before, it's not in the plans to write "official" docs on how to deploy to a cloud environment for the moment.

**Comment by alexyao2015** [comment]
After some inspection [here](https://www.n0r1sk.com/post/running-play-with-docker-on-aws/), it appears that the config you are looking for is [this](https://github.com/play-with-docker/play-with-docker/blob/27d6a43813ca8d19789fbdd027dee0302ab07dff/config/config.go#L68).
